api/v1alpha1/releasestrategy_types.go

```diff
@@ -22,15 +22,32 @@ import (
 
 // ReleaseStrategySpec defines the desired state of ReleaseStrategy
 type ReleaseStrategySpec struct {
-	// Release Tekton Pipeline to execute
+	// Release Tekton Pipeline to execute; used with bundle resolver;
+	// used with bundle resolver
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
-	// +required
-	Pipeline string `json:"pipeline"`
+	// +optional
+	Pipeline string `json:"pipeline,omitempty"`
 
-	// Bundle is a reference to the Tekton bundle where to find the pipeline
+	// Bundle is a reference to the Tekton bundle where to find the pipeline;
+	// used with bundle resolver
 	// +optional
 	Bundle string `json:"bundle,omitempty"`
 
+	// Git repo URL where the Release Tekton Pipeline to execute is located;
```

Comment:
can you add the kubebuilder validations for these?

Reply:
I guess I could. But for the bundle url we never had one either, so I wonder if we need it.

What kind of patterns would you suggest? Some generic url regexp for the url? What about the `GitRevision` and `GitPathInRepo`? I guess I could go find out what github allows for revision strings (we will restrict it later on, but right now I should allow anything) and filenames and create something based on that, but is it needed/worth it?